docs/polkadot/implementors-guide/subsystems-overview.md

```diff
@@ -0,0 +1,95 @@
+
+- Overseer
+- Candidate Selection
+- Candidate Validation
+- Candidate Backing
+- Statement Distribution
+- Misbehavior Arbitration
+
+## Overseer
+
+The overseer is the postal service: it relays messages between the other modules. At any time in this document when we state that a module sends a message to another, it does so via the overseer. The overseer also alerts other subsystems when they should be prepared to work on new relay parents, and when they should shut down preparations to work on old, finalized, or otherwise obsolete relay parents.
+
+## Candidate Selection
+
+The candidate selection module is in charge of seconding 0 or 1 parablock candidates per relay parent.
+
+### Inbound Messages
+
+- **Overseer**: a new parablock candidate has arrived from a collator.
+
+### Outbound Messages
+
+- **Candidate Backing**: this candidate (by hash) satisfies system properties; please validate it
+
+## Candidate Backing
+
+The candidate backing module is in charge of producing statements which assert the (non-)validity of a parachain candidate.
+
+### Inbound Messages
+
+- **Candidate Selection**: a candidate which satisfies system properties
+- **Candidate Validation**: a particular candidate is valid, or not
+
+### Outbound Messages
+
+- **Candidate Validation**: is this candidate in fact valid?
+- **Statement Distribution**: I testify that this candidate is (in-)valid
+
+## Candidate Validation
+
+The candidate validation module handles the details of validating a candidate. For any given parachain candidate, it arranges the appropriate context: the validation function, the relay parent head data, the in-progress relay data, etc.
+
+### Inbound Messages
+
+- **Various Subsystems**: requests to validate a particular parablock candidate
+
+### Outbound Messages
+
+Responses from the Candidate Validation subsystem always return to the requester
+
+- **Various Subsystems**: a parablock candidate is valid, or not
+
+## Statement Distribution
+
+The statement distribution module sends statements to peer nodes, detects double-voting, and tabulates when a sufficient portion of the validator set has unanimously judged a candidate. When judgment is not unanimous, it escalates the issue to misbehavior arbitration.
+
+### Inbound Messages
+
+- **Overseer**: a peer node has seconded a candidate
+
+### Outbound Messages
+
+- **Candidate Validation**: double-check this peer's seconded candidate
+- **Peer validators**: Here's what I think about a candidate's validity
+- **Misbehavior Arbitration**: I disagree with a peer node about this candidate's validity
+- **Overseer**: a unanimous quorum of nodes has agreed about this candidate's validity
+
+## Misbehavior Arbitration
+
+The misbehavior arbitration module kicks in when two validator nodes disagree about a candidate's validity. In this case, _all_ validators, not just those assigned to its parachain, weigh in on the validity of this candidate. The minority is slashed.
```

Comment:
Why not have a specific misbehavior system for double-votes? The dispute between valid/invalid would be handled mostly by the `Validity` module, a runtime API, and a `SecondaryChecking` subsystem.
1. Validity module tracks blocks currently up for approval, backing, output of all secondary checks so far. it handles slashing when there are disputes. 
2. Runtime API makes all information available to node-side
3. SecondaryChecks uses information about blocks up for approval (random seed, escalation) to schedule and perform secondary checks.

Reply:
We always envisioned the validity module counting the parachain validator's backing votes btw, so yes maybe it could just handle their slashing along with its own.

Reply:
This should not be a module, but a subsystem within the validity module; I misspoke. 

The point of Misbehavior Arbitration is that this role handles things when the whole voting set needs to vote on something: this is the subsystem which says "hey, peers: everyone judge this particular block", tallies the votes, and slashes the minority. That behavior is fairly complicated, so I thought it made more sense to put it into its own subsystem than to build it into a different subsystem with a different primary responsibility. 

In other words, assuming I understand the three-step checklist properly, I think Misbehavior Arbitration is the subsystem within the Validity module which handles step 1.